package/src/completionCacheManager/completionCacheManager.ts

```diff
@@ -0,0 +1,23 @@
+import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
+import * as ls from 'vscode-languageserver-types';
+
+export const createCompletionCacheManager = (getFromLanguageService: GetFromLanguageService) => {
+    let completionList: ls.CompletionList | undefined;
+    let lastWord: string | undefined;
+
+    return {
+        getCompletionItems: async (word: string | undefined, resource: monaco.Uri, position: ls.Position) => {
+            const isIncluded = word?.includes(lastWord);
+            const shouldGetItems = !lastWord || !word || !isIncluded;
```

Comment:
This took me a bit to read:
- We're checking if `word` is present multiple times
- We pass `lastWord` to `.includes()` when it's `undefined`. This would be a type error if this repo had TypeScript "strict" mode turned on.

```suggestion
            const shouldGetItems = !lastWord || !word || !word.includes(lastWord)
```

Reply:
I will change it